File: dgx/server-node/src/routers/system.js
'use strict';

const express = require('express');
const os = require('os');
const { requireAuth } = require('../auth');
const { getGpuInfo } = require('../utils/gpu');

const router = express.Router();

router.get('/health', (_req, res) => {
  res.json({ status: 'ok', timestamp: new Date().toISOString() });
});

router.get('/system/info', requireAuth, (_req, res) => {
  const totalMem = os.totalmem();
  const freeMem = os.freemem();

  res.json({
    node_version: process.version,
    os: os.type(),
    platform: os.platform(),
    arch: os.arch(),
    release: os.release(),
    cpu_count: os.cpus().length,
    ram: {
      total_mb: Math.floor(totalMem / (1024 * 1024)),
      available_mb: Math.floor(freeMem / (1024 * 1024)),
    },
    ...getGpuInfo(),
  });
});

module.exports = router;

File: dgx/server/routers/system.py
import platform
import sys
from datetime import datetime, timezone

import psutil
from fastapi import APIRouter, Depends

from ..auth import require_auth
from ..utils.gpu import get_gpu_info

router = APIRouter()


@router.get("/health")
async def health():
    return {"status": "ok", "timestamp": datetime.now(timezone.utc).isoformat()}


@router.get("/system/info")
async def system_info(_: None = Depends(require_auth)):
    mem = psutil.virtual_memory()
    return {
        "python_version": sys.version,
        "os": platform.system(),
        "platform": platform.platform(),
        "cpu_count": psutil.cpu_count(logical=True),
        "ram": {
            "total_mb": mem.total // (1024 * 1024),
            "available_mb": mem.available // (1024 * 1024),
        },
        **get_gpu_info(),
    }
